src/api/java/com/minecolonies/api/crafting/CountedIngredient.java

```diff
@@ -0,0 +1,104 @@
+package com.minecolonies.api.crafting;
+
+import com.google.gson.JsonObject;
+import com.minecolonies.api.util.constant.Constants;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.JSONUtils;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.IIngredientSerializer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * An ingredient that can be used in a vanilla recipe to require more than one item in a particular input slot.
+ *
+ * {
+ *     "type": "minecolonies:counted",
+ *     "item": {
+ *         "item": "minecraft:cobblestone"  // could be a tag or something else
+ *     },
+ *     "count": 16
+ * }
+ */
+public class CountedIngredient extends Ingredient
+{
+    public static final ResourceLocation ID = new ResourceLocation(Constants.MOD_ID, "counted");
+
+    @NotNull
+    private final Ingredient child;
+    private final int count;
+    private ItemStack[] array = null;
+
+    public CountedIngredient(@NotNull final Ingredient child, final int count)
+    {
+        super(Arrays.stream(child.getMatchingStacks()).map(SingleItemList::new));
+
+        this.child = child;
+        this.count = count;
+    }
+
+    /** The underlying ingredient. */
+    @NotNull
+    public Ingredient getChild() { return this.child; }
+
+    /** The number of items required. */
+    public int getCount() { return this.count; }
+
+    @NotNull
+    @Override
+    public ItemStack[] getMatchingStacks()
+    {
+        if (this.array == null)
+        {
+            final List<ItemStack> matchingStacks = Arrays.stream(this.child.getMatchingStacks())
+                    .map(ItemStack::copy).collect(Collectors.toList());
+            matchingStacks.forEach(s -> s.setCount(this.count));
+            this.array = matchingStacks.toArray(new ItemStack[matchingStacks.size()]);
+        }
+        return this.array;
+    }
+
+    @NotNull
+    @Override
+    public IIngredientSerializer<? extends Ingredient> getSerializer()
+    {
+        return Serializer.INSTANCE;
+    }
+
+    public static class Serializer implements IIngredientSerializer<CountedIngredient>
+    {
+        public static final Serializer INSTANCE = new Serializer();

```

Comment:
Again use a getter here for the singleton.
Additionally this is missing the private constructor.

Reply:
Again, this is the standard style.  And it is stateless so there is no harm in allowing public construction; the singleton is for convenience rather than required for correctness.